lib/Doctrine/DBAL/Types/WellKnownTextConverter.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\DBALException;
+
+/**
+ * Does working with this column require SQL conversion functions?
+ *
+ * This is a metadata function that is required for example in the ORM.
+ * Usage of {@link convertToDatabaseValueSQL} and
+ * {@link convertToPHPValueSQL} works for any type and mostly
+ * does nothing. This method can additionally be used for optimization purposes.
+ *
+ * @author Roman Borschel <roman@code-factory.org>
+ * @author Benjamin Eberlei <kontakt@beberlei.de>
+ * @since  2.0
+ */
+interface WellKnownTextConverter
```

Comment:
At first, I named that `RequiresSqlConversion`, tell me what you prefer, or if you'd like something else.

Reply:
I don't really like this name but don't have better proposal ATM, let's think about it...

Reply:
Where does "WellKnown" come from? I don't understand that interface name.

Reply:
If all types extend the base `Type` class, and it has default implementations for the conversion methods, why do we need an additional interface? The caller doesn't need to know if any conversion is happening within the type implementation, it can just call `convert()` and let the type take care of the conversion.

I'm not an ORM user, so could you explain briefly what problem is being solved?

Reply:
@SenseException Ctrl+F "well-know" in the link I gave

Reply:
@morozov this PR is pedantic; it does not solve any issue, it's just about architectural beauty: if you have a boolean method + methods that should be taken into account only if the boolean method returns true, it just begs to be rewritten with an interface. Note that this will make the `Type` class less long with 3 methods that don't do anything removed.

Reply:
But in any case, the caller should either call a method on the object to see if it implements the WKT representation, or check if it implements the interface? The second is even worse because it has to know about two interfaces (`Type` and `WKT`) instead of one (`Type`).

Is it possible to avoid having the caller call this "marker" method at all and just expect it call the `convert()` method instead?

I don't see beauty in making the Type interface smaller because there's still type-related logic which is implemented on the caller side.

Reply:
@morozov Please have a look at #2841, the motivation behind this should be clearer.
Basically it's just about killing `Type` god object, splitting it into smaller pieces and also making consuming Type classes more lightweight (99% of types don't even need this typeof conversion at all, so why would they need to have dummy methods).